src/java.base/windows/classes/sun/nio/ch/PipeImpl.java

```diff
@@ -205,8 +215,16 @@ public SinkChannelImpl sink() {
         return sink;
     }
 
-    private static ServerSocketChannel createListener() throws IOException {
-        ServerSocketChannel listener = ServerSocketChannel.open();
+    private static ServerSocketChannel createListener(boolean preferUnixDomain) throws IOException {
+        ServerSocketChannel listener;
+        if (preferUnixDomain && UnixDomainSockets.isSupported()) {
+            try {
+                listener = ServerSocketChannel.open(UNIX);
+                listener.bind(null);
```

Comment:
Updating version looks okay, I just wondering if bind can fail, in which case it would leak a socket.

Reply:
bind would only fail if it can't find a random unique name in the temp directory. Probability low, but not zero. It doesn't cost much to check if listener != null and close it.